src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -114,14 +118,27 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
                     {
                         if (!declType.IsErrorType())
                         {
-                            Error(diagnostics, ErrorCode.ERR_NoConvToIDisp, declarationSyntax, declType);
+                            Error(diagnostics, hasAwait ? ErrorCode.ERR_NoConvToIAsyncDisp : ErrorCode.ERR_NoConvToIDisp, declarationSyntax, declType);
                         }
 
                         hasErrors = true;
                     }
                 }
             }
 
+            if (!hasErrors && hasAwait)
+            {
+                TypeSymbol taskType = this.Compilation.GetWellKnownType(WellKnownType.System_Threading_Tasks_Task);
+                hasErrors = ReportUseSiteDiagnostics(taskType, diagnostics, _syntax.AwaitKeyword);
+
+                if (!hasErrors)
+                {
+                    var syntaxForAwait = (SyntaxNode)expressionSyntax ?? declarationSyntax;
+                    BoundExpression placeholder = new BoundAwaitableValuePlaceholder(syntaxForAwait, taskType).MakeCompilerGenerated();
+                    (awaitOpt, hasErrors) = BindAwaitInfo(placeholder, syntaxForAwait, diagnostics);
```

Comment:
>(awaitOpt, hasErrors) = BindAwaitInfo(placeholder, syntaxForAwait, diagnostics); [](start = 20, length = 80)

It might still worth doing all this regardless whether we already have errors if ```taskType``` is not an error type. We might get more useful errors, they are not likely to be caused by previous errors, and we will get symbols to expose by SemanticModel/IOperation in the future.